src/SampleWebApp/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/NoxServiceCollectionExtension.g.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Generated
+
+#nullable enable
+
+using Microsoft.EntityFrameworkCore;
+using Nox;
+using Nox.DatabaseProvider;
+using Nox.DatabaseProvider.SqlServer;
+using Nox.Types.EntityFramework.SqlServer;
+using Nox.Types.EntityFramework.vNext;
+using SampleWebApp.Infrastructure.Persistence;
+
+public static class NoxServiceCollectionExtension
+{
+    public static IServiceCollection AddNox(this IServiceCollection services)
+    {
+        services.AddNoxLib();
+        services.AddSingleton<DbContextOptions<SampleWebAppDbContext>>();
+        services.AddSingleton<INoxDatabaseConfigurator, SqlServerDatabaseConfigurator>();
+        services.AddSingleton<INoxDatabaseProvider, SqlServerDatabaseProvider>();
+        services.AddDbContext<SampleWebAppDbContext>();
+        var tmpProvider = services.BuildServiceProvider();
+        var dbContext = tmpProvider.GetRequiredService<SampleWebAppDbContext>();
+        dbContext.Database.EnsureCreated();
```

Comment:
The EnsureCreated, I guess we need to define how each user will run the migrations , etc... should we give complete control to the user on this part?